src/integrationTest/cloudformation/templateRegistryManager.test.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as path from 'path'
+import * as vscode from 'vscode'
+
+import { CloudFormationTemplateRegistry } from '../../shared/cloudformation/templateRegistry'
+import { CloudFormationTemplateRegistryManager } from '../../shared/cloudformation/templateRegistryManager'
+import { rmrf } from '../../shared/filesystem'
+import { getLogger, setLogger } from '../../shared/logger/logger'
+import { makeSampleSamTemplateYaml, strToYamlFile } from '../../test/shared/cloudformation/cloudformationTestUtils'
+import { TestLogger } from '../../test/testLogger'
+import { getTestWorkspaceFolder } from '../integrationTestsUtilities'
+
+/**
+ * Note: these tests are pretty shallow right now. They do not test the following:
+ * * Adding/removing workspace folders
+ */
+describe.only('CloudFormation Template Registry Manager', async () => {
+    let registry: CloudFormationTemplateRegistry
+    let manager: CloudFormationTemplateRegistryManager
+    let workspaceDir: string
+    let testDir: string
+    const additionalFiles: string[] = []
+
+    before(async function() {
+        try {
+            getLogger()
+        } catch (e) {
+            setLogger(new TestLogger())
+        }
+        workspaceDir = getTestWorkspaceFolder()
+        testDir = path.join(workspaceDir, 'cloudFormationTemplateRegistry')
+    })
+
+    beforeEach(() => {
+        registry = new CloudFormationTemplateRegistry()
+        manager = new CloudFormationTemplateRegistryManager(registry)
+    })
+
+    afterEach(async () => {
+        while (additionalFiles.length > 0) {
+            const file = additionalFiles.pop()
+            if (file) {
+                await rmrf(file)
+            }
+        }
+        manager.dispose()
+    })
+
+    it('adds initial template files with yaml and yml extensions at various nesting levels', async () => {
+        await manager.addTemplateGlob('**/test.{yaml,yml}')
+        await new Promise(resolve => setTimeout(resolve, 500))
+        assert.strictEqual(registry.registeredTemplates.length, 2)
```

Comment:
To reduce false negatives, consider looping with a small sleep until the success criteria is met. This can also minimize the time necessary to run the test. Mocha will kill the test if it exceeds the timeout limit. Comment applies to whole file.

```suggestion
        while (registry.registeredTemplates.length !== 2) {
            await new Promise(resolve => setTimeout(resolve, 20))
        }
```

Reply:
Looking at the rest of the file, you may want to put that into an `async waitForTemplateCount` method